server/controllers/users.js

```diff
@@ -1,22 +1,49 @@
+import Joi from '@hapi/joi';
 import { ObjectId } from 'mongodb';
 import User from '../models/User';
 
+import UserSchema from '../validations/users';
+import joiOptions from '../validations/joiOptions';
+
 async function getUser(req, res) {
   try {
     const { userId } = req.params;
     if (!ObjectId.isValid(userId)) {
       return res.status(400).json({ success: false, message: 'Invalid user id value' });
     }
-    const user = await User.findById(userId).select('name email profilePic gender DOB');
+    const user = await User.findById(userId).select('name email profilePic gender DOB').lean();
```

Comment:
What does this `lean()` do?

Reply:
removes unnecessary mongoose code, so you get only the data you need. it should not be used if we are doing any mongo computation on the data, we are directly returning the data.